### ISSUE TITLE
HCA decrypt result inconsistent with libcgss

### ISSUE THREAD

**Issue by nulla2011** [OpenMLTD/MLTDTools#42]
There's a little bit difference between the wave file extracted by HcaDec.exe in this repository and the wave file extracted by hca2wav.exe in [libcgss](https://github.com/hozuki/libcgss) . I put them in two tracks , invert the phase of one of them and found the differences is mostly above 7kHz frequency.

![image](https://user-images.githubusercontent.com/5034243/94364537-8ea70080-00fc-11eb-80cc-8c0b68a93d88.png)

So which one is correct ?

btw , this project's decrypt result is the same with [DereTore](https://github.com/OpenCGSS/DereTore) .

**Comment by hozuki** [comment]
Yes MLTDTools still uses DereTore's decoder, which is a direct port of the original decoder in C++.
libcgss integrated some of Ishotihadus' improvements (see his [repo](https://github.com/Ishotihadus/hca)).
So... I guess Isho's is correct in high frequency.

**Comment by nulla2011** [comment]
OK. I made a comparation. The result of libcgss' latest version is the closest to the CD version. 39.
But I found it is almost monaural in high frequency, I dont know why.

**Comment by hozuki** [comment]
BTW for a mathematically correct encoder/decoder please refer to [VGAudio](https://github.com/Thealexbarney/VGAudio).
